modules/registry/registry-core/src/main/java/org/apache/airavata/registry/core/app/catalog/model/UserResourceProfilePK.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.airavata.registry.core.app.catalog.model;
+
+import java.io.Serializable;
+
+public class UserResourceProfilePK implements Serializable {
+
+    private String userId;
+
+    private String gatewayID;
+
+    public UserResourceProfilePK(String userId, String gatewayID) {
+        this.userId = userId;
+        this.gatewayID = gatewayID;
+    }
+
+    public UserResourceProfilePK() {
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why is hashCode and equals implemented like this? Two UserResourceProfilePK instances with the same userId and gatewayID should be equal and should hash to the same value, right?